vsphere/stackstate_checks/vsphere/vsphere.py

```diff
@@ -1066,29 +1070,31 @@ def _vsphere_datastores(self, content, domain="Unspecified", regexes=None, inclu
             topology_tags = {}
             labels = []
             hostname = c.name
+            try:
+                if isinstance(c, vim.Datastore):
+                    topology_tags["topo_type"] = VSPHERE_COMPONENT_TYPE.DATASTORE
+                    topology_tags["name"] = c.name
+                    topology_tags["layer"] = TOPOLOGY_LAYERS.DATASTORE
+                    topology_tags["domain"] = domain
+                    topology_tags["accessible"] = c.summary.accessible
+                    topology_tags["capacity"] = str(c.summary.capacity)
+                    topology_tags["type"] = c.summary.type
+                    topology_tags["url"] = c.summary.url
 
-            if isinstance(c, vim.Datastore):
-                topology_tags["topo_type"] = VSPHERE_COMPONENT_TYPE.DATASTORE
-                topology_tags["name"] = c.name
-                topology_tags["accessible"] = c.summary.accessible
-                topology_tags["capacity"] = str(c.summary.capacity)
-                topology_tags["type"] = c.summary.type
-                topology_tags["url"] = c.summary.url
-                topology_tags["layer"] = TOPOLOGY_LAYERS.DATASTORE
-                topology_tags["domain"] = domain
-
-                sts_identifiers, labels = self.extract_tags("Datastore", c._moId)
-                topology_tags["identifiers"] = sts_identifiers
+                    sts_identifiers, labels = self.extract_tags("Datastore", c._moId)
```

Comment:
Suggest you group the required attributes first and collect others (starting from `accessible`) later so that we have a complete component even if fetching `accessible` or `capacity` fails.